### PR TITLE
Support `import Mod, only: :sigils`

### PR DESCRIPTION
**Pull request by wojtekmach**
Ref https://groups.google.com/d/msg/elixir-lang-core/QGU3ARQeL7w/AJMEtsYgBQAJ

### CODE REVIEW

File: lib/elixir/src/elixir_import.erl [hunk 1]
```diff
@@ -29,6 +35,16 @@ import(Meta, Ref, Opts, E) ->
   elixir_env:trace({import, [{imported, Added} | Meta], Ref, Opts}, E),
   {Functions, Macros}.
 
+take_sigils(List) ->
+  lists:filter(fun is_sigil/1, List).
+
+is_sigil({Name, Arity}) ->
+  Arity == 2 andalso is_sigil(atom_to_list(Name));
+is_sigil("sigil_" ++ Rest) ->
+  lists:all(fun(X) -> (X >= $a andalso X =< $z) orelse (X >= $A andalso X =< $Z) end, Rest);
```
Comment (josevalim) [inline lib/elixir/src/elixir_import.erl]: This definition is too loose at the moment. :)

File: lib/elixir/src/elixir_import.erl [hunk 2]
```diff
@@ -12,8 +12,14 @@ import(Meta, Ref, Opts, E) ->
...
-        {Added2, Macs} = import_macros(true, Meta, Ref, Opts, E),
-        {keydelete(Ref, ?key(E, functions)), Macs, Added2};
+        {Added1, Macs} = import_macros(true, Meta, Ref, Opts, E),
+        {keydelete(Ref, ?key(E, functions)), Macs, Added1};
+      {only, sigils} ->
+        {Added1, Funs1} = import_functions(Meta, Ref, Opts, E),
+        {Added2, Macs1} = import_macros(false, Meta, Ref, Opts, E),
+        Funs2 = keyupdate(Ref, fun take_sigils/1, Funs1),
+        Macs2 = keyupdate(Ref, fun take_sigils/1, Macs1),
+        {Funs2, Macs2, Added1 or Added2};
```
Comment (fertapric) [inline lib/elixir/src/elixir_import.erl]: I believe the trace will return `imported: true` even if no sigils are imported, and I think the trace in that case (if no sigils are imported) should be `imported: false`. In other words, I think `Added1 or Added2` needs to be calculated after filtering the sigils.

File: lib/elixir/test/elixir/kernel/import_test.exs [hunk 3]
```diff
@@ -137,6 +137,22 @@ defmodule Kernel.ImportTest do
     import String
   end
 
+  defmodule Sigils do
+    # when imported it should cause conflict
+    def is_integer(_), do: false
+
+    def sigil_X(_, _), do: :x
+
+    defmacro sigil_Y(_, _), do: :y
```
Comment (josevalim) [inline lib/elixir/test/elixir/kernel/import_test.exs]: Let's define a fake sigil too, `sigil__` and make sure it is not imported. You can check if something is imported by looking inside `__ENV__`.

**Comment by fertapric** [comment]
Thanks @wojtekmach! 💜

**Pull request merged**